Travis CI seems to fail at various locations in `moveit_joy.py`. Still not sure why.
- https://travis-ci.org/tork-a/rtmros_nextage/jobs/139635943#L6107
  
  ```
  -- run_tests.py: execute commands
  /opt/ros/indigo/share/rostest/cmake/../../../bin/rostest --pkgdir=/home/travis/catkin_ws/src/rtmros_nextage/nextage_moveit_config --package=nextage_moveit_config --results-filename test_joy.xml --results-base-dir /home/travis/catkin_ws/build/nextage_moveit_config/test_results /home/travis/catkin_ws/src/rtmros_nextage/nextage_moveit_config/test/joy.test 
  ... logging to /home/travis/.ros/log/rostest-testing-gce-f3a1d1d0-6097-4c88-9be6-773263d22f23-27299.log
  ... logging to /home/travis/.ros/log/rostest-testing-gce-f3a1d1d0-6097-4c88-9be6-773263d22f23-27298.log
  [ROSUNIT] Outputting test results to /home/travis/catkin_ws/build/nextage_moveit_config/test_results/nextage_moveit_config/rostest-test_joy.xml
  Traceback (most recent call last):
  File "/opt/ros/indigo/lib/moveit_ros_visualization/moveit_joy.py", line 44, in <module>
    from moveit_ros_planning_interface._moveit_robot_interface import RobotInterface
  KeyboardInterrupt
  ```
- https://travis-ci.org/tork-a/rtmros_nextage/jobs/139834751#L6114
  
  ```
  [ROSUNIT] Outputting test results to /home/travis/catkin_ws/build/nextage_moveit_config/test_results/nextage_moveit_config/rostest-test_joy.xml
  Traceback (most recent call last):
  File "/opt/ros/indigo/lib/moveit_ros_visualization/moveit_joy.py", line 43, in <module>
    import threading
  File "/usr/lib/python2.7/threading.py", line 1, in <module>
    """Thread module emulating a subset of Java's threading model."""
  KeyboardInterrupt
  [ROSUNIT] Outputting test results to /home/travis/catkin_ws/build/nextage_moveit_config/test_results/nextage_moveit_config/rostest-test_nxo_moveit.xml
  omniORB: Failed to bind to address :: port 2809. Address in use?
  omniORB: Error: Unable to create an endpoint of this description: giop:tcp::2809
  Failed to initialise the ORB / POA: INITIALIZE_TransportError
  Is omniNames already running?
  testjoy_topic_hz ... FAILURE!
  FAILURE: Test Fixture Nodes ['joy'] failed to launch
  File "/usr/lib/python2.7/unittest/case.py", line 331, in run
    testMethod()
  File "/opt/ros/indigo/lib/python2.7/dist-packages/rostest/runner.py", line 121, in fn
    self.assert_(not failed, "Test Fixture Nodes %s failed to launch"%failed)
  File "/usr/lib/python2.7/unittest/case.py", line 424, in assertTrue
    raise self.failureException(msg)
  --------------------------------------------------------------------------------
  [ROSTEST]-----------------------------------------------------------------------
  [testjoy_topic_hz][failed]
  ```

--- a/nextage_moveit_config/test/joy.test
+++ b/nextage_moveit_config/test/joy.test
@@ -1,15 +1,44 @@
 <launch>
-  <!-- Better test desired. -->
-  <include file="$(find nextage_moveit_config)/launch/joystick_control.launch" />
+  <!-- moveit_joy.py requires /move_group -->
+  <!-- follow the tutorials on  http://wiki.ros.org/rtmros_nextage/Tutorials/Programming_Hiro_NEXTAGE_OPEN_GUI
+       moveit_planning_execution.launch starts rviz so we run launch files except moviet_rviz.launch -->
+  <!-- <include file="$(find nextage_moveit_config)/launch/moveit_planning_execution.launch"> -->
+ <arg name="kinematics_conf" default="$(find nextage_moveit_config)/config/kinematics_kdl.yaml" />
+ <!-- for debug -->
+ <!-- <include file="$(find nextage_moveit_config)/launch/moveit_rviz.launch"> -->
+ <!--    <arg name="config" value="true"/> -->
+ <!--    <arg name="kinematics_conf" value="$(arg kinematics_conf)"/> -->
+ <!-- </include> -->
+  <include file="$(find nextage_moveit_config)/launch/move_group.launch">
+    <arg name="publish_monitored_planning_scene" value="true" />
+  </include>
+  <include file="$(find nextage_moveit_config)/launch/planning_context.launch">
+    <arg name="load_robot_description" value="true"/>
+  </include>
+  <!-- -->
 
-  <!-- Check if MoveGroup is running -->
-  <arg name='TESTNAME_JOY_STATUS' value='joy_topic_hz' />
-  <test pkg="rostest" type="hztest" test-name="$(arg TESTNAME_JOY_STATUS)" name="$(arg TESTNAME_JOY_STATUS)" time-limit="60">
-    <param name="topic" value="/joy" />
-    <param name="hz" value="40.0" />
-    <param name="hzerror" value="2.00" />
-    <param name="test_duration" value="10.0" />
-    <param name="wait_time" value="15.0" />
+  <!-- publish dummy update_full -->
+  <node pkg="rostopic" type="rostopic" name="publish_update_full"
+        args="pub -r 1 /rviz_moveit_motion_planning_display/robot_interaction_interactive_marker_topic/update_full visualization_msgs/InteractiveMarkerInit '{markers: [{header: {frame_id: /world}, name: &quot;EE:goal_LARM_JOINT5_Link&quot;}]}'" />
+  <!-- Publish dummy /joy moveit_joy need to be button push action : https://github.com/ros-planning/moveit_ros/blob/c6d0edea532048ae7036502ae8f464ae0f92ebb8/visualization/python/moveit_ros_visualization/moveit_joy.py#L585-->
+  <node pkg="rostopic" type="rostopic" name="publish_joy_topic_off"
+        args="pub -r 2 /joy sensor_msgs/Joy '{header: {frame_id: 'WAIST'}, axes: [0,0,0,0,0, 0,0,0],
+             buttons:[0,0,0,0,0, 0,0,0,0,0, 0]}'" />
+  <node pkg="rostopic" type="rostopic" name="publish_joy_topic_on"
+        args="pub -r 3 /joy sensor_msgs/Joy '{header: {frame_id: 'WAIST'}, axes: [0,0,0,0,0, 0,0,0],
+             buttons:[0,0,1,0,0, 0,0,0,0,0, 0]}'" />
+
+  <!-- Launch python interface -->
+  <node pkg="moveit_ros_visualization" type="moveit_joy.py" output="screen" name="moveit_joy"/>
+
+  <!-- Check if moveit_joy is running -->
+  <arg name='TESTNAME_MOVEPLAN_JOY' value='hz_moveplan_joy' />
+  <test pkg="rostest" type="hztest" test-name="$(arg TESTNAME_MOVEPLAN_JOY)" name="$(arg TESTNAME_MOVEPLAN_JOY)" time-limit="60">
+    <param name="topic" value="/rviz/moveit/plan" />
+    <param name="hz" value="5.0" />
+    <param name="hzerror" value="5.00" />
+    <param name="test_duration" value="1.0" />
+    <param name="wait_time" value="10.0" />
   </test>  
          
 </launch>

